src/wp-includes/fonts.php

```diff
@@ -107,31 +125,73 @@ function wp_unregister_font_collection( string $slug ) {
  *     @type string|false $error   False or error message.
  * }
  */
-function wp_get_font_dir() {
+function wp_font_dir( $create_dir = true ) {
+	/*
+	 * Allow extenders to manipulate the font directory consistently.
+	 *
+	 * Ensures the upload_dir filter is fired both when calling this function
+	 * directly and when the upload directory is filtered in the Font Face
+	 * REST API endpoint.
+	 */
+	add_filter( 'upload_dir', 'wp_apply_font_dir_filter' );
+	$font_dir = wp_upload_dir( null, $create_dir, false );
+	remove_filter( 'upload_dir', 'wp_apply_font_dir_filter' );
+	return $font_dir;
+}
+
+/**
+ * Applies filters to manipulate the font directory used by the font library.
+ *
+ * Wrapper for apply_filters( 'font_dir', $font_dir ).
+ * Does not apply filters if already doing the {@see 'font_dir'} filter.
+ *
+ * Allows developers to manipulate the font directory used by the
+ * font library.
+ *
+ * @since 6.5.0
+ *
+ * @param string $font_dir The font directory.
+ * @return string The modified font directory.
+ */
+function wp_apply_font_dir_filter( $font_dir ) {
+	if ( doing_filter( 'font_dir' ) ) {
+		// Avoid an infinite loop.
+		return $font_dir;
+	}
+
+	return apply_filters( 'font_dir', $font_dir );
+}
+
+/**
+ * Modifies the default font directory.
+ *
+ * This function should not be called directly, use wp_get_font_dir() instead.
+ *
+ * Modifies the default font directory used by WordPress to the subdirectory fonts
+ * within the content directory. This function should not be called directly but
+ * rather applied to the {@see 'font_dir'} hook.
+ *
+ * @access private
+ * @since 6.5.0
+ *
+ * @see wp_get_font_dir()
+ *
+ * @return array See wp_font_dir() for description.
+ */
+function wp_default_font_dir_filter() {
```

Comment:
Could this function be called `wp_default_font_dir` instead? The `_filter` part seems unnecessary, and I haven't seen any other core function called in that fashion.

Reply:
Done in 1c870cc5dd1b4a97253eb062f0127f6bcc5c8050